p2719.md

```diff
@@ -189,67 +194,69 @@ or without a `size_t` parameter depending on whether the considered `operator de
-```c++
-operator new(sizeof(T), std::type_identity<T>{}, placement-args...)
-```
-In particular, this means that it triggers ADL, which allows defining such an operator in e.g. a custom namespace.
-We mostly discuss `operator new` below, but `operator new[]`, `operator delete` and `operator delete[]` are all
-handled analogously.
-
-## Wording approach
-
-This proposal is best explained as two distinct changes to the existing rules explained above. First, observe how we
-currently perform name lookup first to fix the name we're using (either `T::operator new` or `::operator new`), and
-then perform overload resolution to find the best candidate given that name. This needs to change because if we introduce
-a free function in the mix that is found by ADL, we could often end up finding that free function as the result of name
-lookup. However, the mere presence of such a free function doesn't guarantee at all that it is a viable candidate given
-the arguments we have (for example if the free function merely happened to be in one of the namespaces we looked in). As
-a result, we could often end up in a situtation where `new Foo(...)` fails during overload resolution simply because a
-non-viable `operator new` free function was located in one of `Foo`'s associated namespaces.
-
-Consequently, the first change we need to make is to perform name lookup for `T::operator new`, and if found, then perform
-overload resolution on that name immediately. If overload resolution fails, then go on to the next candidate (which is
-`::operator new`) and perform overload resolution on that name. This change can be seen as a relaxation of the current
-rules and cannot affect any existing valid program. Indeed, code where `T::operator new` is currently selected but where
-overload resolution fails actually doesn't compile today, since the compiler falls flat after failing to perform overload
-resolution. After this change, such code would now fall back to trying overload resolution on `::operator new`, which might
-succeed.
-
-The second change we need to make is the addition of a new step in the search, between `T::operator new` and `::operator new`.
-Assuming that the name lookup or the overload resolution for `T::operator new` fails, we would now perform an argument-dependent
-name lookup for a free function named `operator new` as-if we had the following expression:
+This proposal adds a new implicit tag argument of type `type_identity_t<T>` to `operator new` and `operator delete` that is incorporated into the existing declaration preference logic with a higher priority than existing implicit parameters. To avoid potential conficts with existing code, this parameter is placed as the first argument to the operator, preceding the size or subject pointer. For such a declaration to be valid we require that the parameter be explicitly a specialization of `std::type_identity_t`, and not a fully dependent type.
```

Comment:
```suggestion
This proposal adds a new implicit tag argument of type `type_identity<T>` to `operator new` and `operator delete` that is incorporated into the existing declaration preference logic with a higher priority than existing implicit parameters. To avoid potential conficts with existing code, this parameter is placed as the first argument to the operator, preceding the size or subject pointer. For such a declaration to be valid we require that the parameter be explicitly a specialization of `std::type_identity`, and not a fully dependent type.
```